src/components/animated-progress.tsx

```diff
@@ -0,0 +1,181 @@
+const pacmanBottomFramesAlternate = keyframes`
+  0% { transform: rotateZ(0.1deg) }
+  100% { transform: rotateZ(30deg) }
+`;
+interface PacmanBodyProps {
+  color: string;
+  pacmanSize: number;
+  alternate: boolean;
+}
+const PacmanBodyTop = styled.div<PacmanBodyProps>`
+  position: absolute;
+  top: 0;
+  height: ${({ pacmanSize }) => pacmanSize / 2}px;
+  width: ${({ pacmanSize }) => pacmanSize}px;
+  background: ${({ color }) => color};
+  border-top-left-radius: ${({ pacmanSize }) => pacmanSize / 2}px;
+  border-top-right-radius: ${({ pacmanSize }) => pacmanSize / 2}px;
+  // NOTE: So the top and bottom always overlap when sizes are in decimals.
+  box-shadow: 0 0 0 0.1px ${({ color }) => color};
+  animation-name: ${({ alternate }) =>
+    alternate ? pacmanTopFrames : pacmanTopFramesAlternate};
+  animation-duration: 0.12s;
+  animation-timing-function: linear;
+  animation-iteration-count: 10;
+  animation-direction: alternate;
+  animation-fill-mode: both;
+`;
+const PacmanBodyBottom = styled(PacmanBodyTop)`
+  top: 50%;
+  border-top-left-radius: 0;
+  border-top-right-radius: 0;
+  border-bottom-left-radius: ${({ pacmanSize }) => pacmanSize / 2}px;
+  border-bottom-right-radius: ${({ pacmanSize }) => pacmanSize / 2}px;
+  animation-name: ${({ alternate }) =>
+    alternate ? pacmanBottomFrames : pacmanBottomFramesAlternate};
+`;
+
+export type AnimatedProgressProps = {
+  color?: string;
+  size?: number;
+  pacmanColor?: string;
+};
+
+const AnimatedProgress = forwardRef<HTMLDivElement, AnimatedProgressProps>(
+  ({ color: circleColor = '#fff', size: circleSize = 7.5, ...props }, ref) => {
+    const {
+      slideCount,
+      skipTo,
+      activeView: { slideIndex }
+    } = useContext(DeckContext);
+
+    const [pacmanOffsetLeft, setPacmanOffsetLeft] = useState<number | null>(
+      null
+    );
+    const [pacmanOffsetTop, setPacmanOffsetTop] = useState<number | null>(null);
+    const [alternateAnimation, setAlternateAnimation] = useState(false);
+
+    const [activeCircleNode, setActiveCircleNode] =
+      useState<HTMLDivElement | null>(null);
+    const activeCircleCallbackRef = useCallback(
+      (activeCircleNode: HTMLDivElement) => {
+        setActiveCircleNode(activeCircleNode);
+      },
+      []
+    );
+
+    useEffect(() => {
+      if (activeCircleNode?.offsetParent) {
+        const { offsetLeft, offsetTop } = activeCircleNode;
+        const halfOfCircleOccupiedSpace =
+          circleSize / 2 + PROGRESS_CIRCLE_BORDER_WIDTH;
+        setPacmanOffsetLeft(offsetLeft + halfOfCircleOccupiedSpace);
+        setPacmanOffsetTop(offsetTop + halfOfCircleOccupiedSpace);
+        setAlternateAnimation((alternateAnimation) => !alternateAnimation);
+      } else {
+        setPacmanOffsetLeft(null);
+        setPacmanOffsetTop(null);
+      }
+    }, [circleSize, activeCircleNode]);
+
+    const circleMargin = circleSize / 1.64;
```

Comment:
I'd love to just hoist or note what this magic number is.

Reply:
👍  Of course. Have improved it by adding the calculation in favour of the magic number here: https://github.com/FormidableLabs/spectacle/pull/1124/commits/5db35bd40b7d51148f607b5ba0740d2911ddb01a